### DISCUSSION TITLE
Can pinbox read from mbox files?

### DISCUSSION THREAD

**Discussion by siliconpi** [msp301/pinbox#42] [Q&A] [answered]
1. I have a large ~30 GB mbox file - can I setup pinbox to read and display emails from there?
2. Will pinbox work fine in "read-only" mode, where the emails can be displayed, printed, searched within, but not deleted or forwarded. It will not be connected to an active account. Is that possible?
3. I need to provide read-only access to that exact mbox file to multiple people. Would pinbox work fine with multiple people accessing it? There is no need to save any "state" for each user, but multiple users might be searching, viewing at the same time. Would pinbox support that use-case?

**Comment by msp301** [comment, marked as answer]
Hi @siliconpi 

To answer your questions:

1. Pinbox uses the [notmuch](https://notmuch.readthedocs.io/en/latest/man1/notmuch.html) to read emails. Their documentation sadly says it cannot read mbox files directly. They have to be extracted into a mail directory before notmuch can read them; this can be done with a tool like [mb2md](https://github.com/dovecot/tools/blob/main/mb2md.pl). See the notmuch setup steps for more information: https://notmuch.readthedocs.io/en/latest/man1/notmuch.html#setup

2. Yes, read-only should be fine. At the moment Pinbox is only capable of reading a mail directory and viewing emails through its web interface. I test Pinbox myself by downloading and extracting a mbox file and keep it completely disconnected and that works.

3. If you'd like several people to access the same mailbox then yes that should work fine too. Please be mindful that Pinbox is insecure on its own, so I would advise keeping it offline and/or secured behind a VPN. No authentication is built in so some consideration is needed for what content is being exposed to who.

It's early days for Pinbox and I've been working on this on my own when I have had time. It's usable but far from feature complete or in a finished state. If you do choose to give Pinbox a try, please consider posting feedback and/or making contributions :)

This is unrelated but I'll post this here as an FYI - I've recently been working on packaging Pinbox into a complete Docker container to help getting something up and running, hopefully I'll have that done before the end of the year. I have also been considering re-licensing Pinbox to AGPL to help promote public contributions in future.

I hope this helps

**Reply by siliconpi** [reply]
Would you mind emailing me - there is a discussion I'll like to have in private if you don't mind. I'm at my username @gmail.com
